components/flow/flow-service/src/main/kotlin/net/corda/flow/application/sessions/utils/SessionUtils.kt

```diff
@@ -14,4 +15,14 @@ internal object SessionUtils {
             throw CordaRuntimeException("Session: $sessionId Status is ${status?.name ?: "NULL"}")
         }
     }
+
+    fun checkPayloadMaxSize(serializedPayload: ByteArray, flowFiberService: FlowFiberService) {
+        val payloadSize = serializedPayload.size
+        val checkpoint = flowFiberService.getExecutingFiber().getExecutionContext().flowCheckpoint
+        val maxPayloadSize = checkpoint.maxPayloadSize
+
+        if (payloadSize > maxPayloadSize) {
+            throw FlowPlatformException("Payload size $payloadSize larger than allowed $maxPayloadSize")
```

Comment:
It might be more clear to add the units
```suggestion
            throw FlowPlatformException("Payload size: $payloadSize bytes larger than allowed: $maxPayloadSize bytes.")
```